app/cms/github-webhook.server.ts

```diff
@@ -14,9 +14,11 @@ type ProcessResult = {
   cacheKeysToInvalidate: Set<string>
 }
 
-export function pushEventCacheKeysToInvalidate(
-  event: PushEvent
-): ProcessResult {
+export type PathsResult = {
+  updatedDocuments: Set<string>
+}
+
+function getMatchingDocCollectionsForPushEvent(event: PushEvent) {
```

Comment:
There's a `getCollectionsForRepo` at `app/cms/doc-collections/get-collections-for-repo.ts`... perhaps we could use that instead here? Just modify it to take an option branch parameter?